File: Year-2022/LeetCode June Challenge/TwoSum2.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> twoSum1(vector<int>& numbers, int target) {
        int start = 0, end = numbers.size() - 1;
        while(numbers[start] + numbers[end] != target){
            if(numbers[start] + numbers[end] < target) start++;
            else end--;
        }
        return {start + 1, end + 1};
    }
    vector<int> twoSum(vector<int>& numbers, int target) {
        int start = 0, end = numbers.size() - 1;
        while(start <= end){
            if(numbers[start] + numbers[end] == target)
                return {start + 1, end + 1};
            else if(numbers[start] + numbers[end] < target)
                start++;
            else end--;
        }
        return {};
    }
};
int main(){
    vector<int> numbers = {1,2,3,4,4,9,56,90};
    int target = 8;
    Solution s;
    vector<int> ans = s.twoSum(numbers, target);
    for(int i = 0; i < ans.size(); ++i) cout<<ans[i]<<" ";
    return 0;
}

File: Year-2022/LeetCode June Challenge/TwoSum2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int start = 0, end = numbers.length - 1;
        while(numbers[start] + numbers[end] != target){
            if(numbers[start] + numbers[end] < target)
                start++;
            else end--;
        }
        int[] ans = {start + 1, end + 1};
        return ans;
    }
}
public class TwoSum2 {
    public static void main(String[] args) {
        int[] numbers = {2,7,11,15};
        int target = 9;
        Solution s = new Solution();
        int[] ans = s.twoSum(numbers, target);
        for(int i : ans){
            System.out.println(i);
        }
    }
}
